plugins/modules/zos_script.py

```diff
@@ -0,0 +1,416 @@
+    in the remote machine.
+
+options:
+  chdir:
+    description:
+      - Change the script's working directory to this path.
+      - When not specified, the script will run in the user's
+        home directory on the remote machine.
+    type: str
+    required: false
+  cmd:
+    description:
+      - Path to the local or remote script followed by optional arguments.
+      - If the script path contains spaces, make sure to enclose it in two
+        pairs of quotes.
+      - Arguments may need to be escaped so the shell in the remote machine
+        handles them correctly.
+    type: str
+    required: true
+  creates:
+    description:
+      - Path to a file in the remote machine. If it exists, the
+        script will not be executed.
+    type: str
+    required: false
+  encoding:
+    description:
+      - Specifies which encodings the script should be converted from and to.
+      - If C(encoding) is not provided, the module determines which local
+        and remote charsets to convert the data from and to.
+    type: dict
+    required: false
+    suboptions:
+      from:
+        description:
+          - The encoding to be converted from.
+        required: true
+        type: str
+      to:
+        description:
+          - The encoding to be converted to.
+        required: true
+        type: str
+  executable:
+    description:
+      - Path of an executable in the remote machine to invoke the
+        script with.
+      - When not specified, the system will assume the script is
+        interpreted REXX and try to run it as such. Make sure to
+        include a comment identifying the script as REXX at the
+        start of the file in this case.
+    type: str
+    required: false
+  remote_src:
+    description:
+      - If set to C(false), the module will search the script in the
+        controller.
+      - If set to C(true), the module will search the script in the
+        remote machine.
+    type: bool
+    required: false
+  removes:
+    description:
+      - Path to a file in the remote machine. If it does not exist, the
+        script will not be executed.
+    type: str
+    required: false
+  tmp_path:
+    description:
+      - Path in the remote machine where local scripts will be
+        temporarily copied to.
+      - When not specified, the module will copy local scripts to
+        the default temporary path for the user.
+      - If C(tmp_path) does not exist in the remote machine, the
+        module will not create it.
+    type: str
+    required: false
+
+extends_documentation_fragment:
+  - ibm.ibm_zos_core.template
+
+notes:
+  - When executing local scripts, temporary storage will be used
+    on the remote z/OS system. The size of the temporary storage will
+    correspond to the size of the file being copied.
+  - Execution permissions for the group assigned to the script will be
+    added to remote scripts. The original permissions for the script will
```

Comment:
Yes, it will only apply to remote scripts, I'll add that clarification to the documentation. Local scripts that were copied to a managed node are deleted before the tasks ends

Reply:
Yes, it will only apply to remote scripts, I'll add that clarification to the documentation. Local scripts that were copied to a managed node are deleted before the tasks ends